src/main/java/com/icthh/xm/tmf/ms/communication/config/KafkaChannelFactory.java

```diff
@@ -114,15 +132,26 @@ public void createHandler() {
         }
 
         channel.subscribe(message -> {
-            try {
-                MdcUtils.putRid(MdcUtils.generateRid());
-                handleEvent(message);
-            } catch (Exception e) {
-                log.error("error processing event", e);
-                throw e;
-            } finally {
-                MdcUtils.removeRid();
+
+            long delay = nextScheduledTime.getAndAdd(pauseBetweenSends) - System.nanoTime() / 1000;
+            if (delay > kafkaReadSleepTimeout) {
+                try {
+                    Thread.sleep(kafkaReadSleepTimeout);
```

Comment:
How about maxPollInterval, what will happend if this sleep time will be more that maxPollInterval?
You consumer will be mark as dead in broker, and every time rebalancing will be invoked.
I think better to poll, then if rate limit disallow to process message just not move offset, sleep for Math.min(maxPollInterval/10, kafkaReadSleepTimeout) and poll again.

Reply:
Hi @sergeysenja1992!
Unfortunately, I didn't find how spring cloud configures this property. 
`KafkaProperties` didn't have this field and also  I checked the consumer field and didn't find it too. I know, in default, spring kafka set max pool time at 5 min. If you can provide more info about where I can find the necessary property,  I'd appreciate it.